simulator/core/simulator.h

```diff
@@ -58,9 +108,41 @@ extern int usetestflash;
 // Simulator state
 extern int cycle;
 #ifdef HAVE_REPLAY
+/** Seek to the specified simulated cycle, if possible.
+ *
+ * This function allows to set the simulator state to a cycle in the past or,
+ * if the cycle state has already been rewound, in the future. It updates the
+ * simulator (and the pipeline if enabled) cycle to the value specified. This
+ * operation can fail if the simulator and pipeline states fall out of sync. If
+ * the target cycle is too far foward, it sets the cycle to the latest possible
+ * cycle the simulator knows.
+ *
+ * @param[in] target Target cycle to seek to.
+ *
+ * @returns True on success, false if it was not able to.
+ * @post On success, the simulator was set to the specified target cycle. On
+ * failure, most likely the pipeline and simulator are no longer in sync.
+ */
 bool simulator_state_seek(int target);
 #endif
+
+// FIXME misnomer
```

Comment:
Wow, yeah, I had to re-read that code a few times to re-grok what this is doing. Should probably be something closer to `sim_update_and_pause_elapsed_time` -- wordy, but clearer..